app/controllers/webauthn_verifications_controller.rb

```diff
@@ -34,13 +38,20 @@ def authenticate
 
   private
 
+  def set_verification
+    @verification = WebauthnVerification.find_by(path_token: webauthn_token_param)
+
+    render_not_found and return unless @verification
+    redirect_to root_path, alert: t(".expired_or_already_used") if @verification.path_token_expires_at < Time.now.utc
+  end
+
   def set_user
-    verification = WebauthnVerification.find_by(path_token: webauthn_token_param)
-    if !verification || verification.path_token_expires_at < Time.now.utc
-      render_not_found
-    else
-      @user = verification.user
-    end
+    @user = @verification.user
+  end
+
+  def expire_webauthn_verification
+    @verification.path_token_expires_at = 1.second.ago
+    @verification.save!
   end
```

Comment:
Would this method make more sense in the webauthn_verifications model?

Reply:
Yes, I think so.